### PR TITLE
[SCB-1130] Fix single point failure of Alpha cluster when scanner enabled node shutdown

### PR DESCRIPTION
**Pull request by coolbeevip**
Dynamically configure the master node base on master_lock table

Configure `alpha.cluster.master.enabled=true` to activate this feature

### Attention
* mysql need configure `serverTimezone=GMT%2b8`
* need configure NTP

### example
Start two nodes
You can see the `Master Node` or `Slave Node` in the log. When the Master Node shutdown, Slave Node change to the Master Node.

```
java -jar alpha-server-0.4.0-SNAPSHOT-exec.jar \
  --server.port=0 \
  --alpha.server.port=8080 \
  --spring.datasource.url="jdbc:postgresql://127.0.0.1:5432/saga?useSSL=false" \
  --spring.datasource.username=saga-user \
  --spring.datasource.password=saga-password \
  --alpha.cluster.master.enabled=true
```

```
java -jar alpha-server-0.4.0-SNAPSHOT-exec.jar \
  --server.port=0 \
  --alpha.server.port=8081 \
  --spring.datasource.url="jdbc:postgresql://127.0.0.1:5432/saga?useSSL=false" \
  --spring.datasource.username=saga-user \
  --spring.datasource.password=saga-password \
  --alpha.cluster.master.enabled=true
```

### CODE REVIEW

**Comment by WillemJiang** [comment]
There are some code which are not tested,  please add some unit tests to verify cluster setting.

File: alpha/alpha-server/src/main/java/org/apache/servicecomb/pack/alpha/server/cluster/master/provider/Locked.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.pack.alpha.server.cluster.master.provider;
+
+public interface Locked {
```
Comment (WillemJiang) [inline alpha/alpha-server/src/main/java/org/apache/servicecomb/pack/alpha/server/cluster/master/provider/Locked.java]: We should use noun as  the interface name.

Comment (coolbeevip) [reply alpha/alpha-server/src/main/java/org/apache/servicecomb/pack/alpha/server/cluster/master/provider/Locked.java]: agree

File: alpha/alpha-core/src/main/java/org/apache/servicecomb/pack/alpha/core/NodeType.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.pack.alpha.core;
+
+public class NodeType {
```
[the remark this answers is not in the document]
Comment (coolbeevip) [reply alpha/alpha-core/src/main/java/org/apache/servicecomb/pack/alpha/core/NodeType.java]: agree

File: alpha/alpha-server/src/main/java/org/apache/servicecomb/pack/alpha/server/cluster/master/provider/AbstractLockProvider.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+
+import org.apache.servicecomb.pack.alpha.server.cluster.master.LockConfig;
+
+import java.util.Optional;
+
+public abstract class AbstractLockProvider implements LockProvider {
+
+    private final LockProviderPersistence lockProviderPersistence;
+    private boolean lockInitialization = Boolean.FALSE;
```
[the remark this answers is not in the document]
Comment (coolbeevip) [reply alpha/alpha-server/src/main/java/org/apache/servicecomb/pack/alpha/server/cluster/master/provider/AbstractLockProvider.java]: fixed

File: alpha/alpha-server/src/main/java/org/apache/servicecomb/pack/alpha/server/AlphaConfig.java [hunk 4]
```diff
@@ -100,7 +110,7 @@ TxConsistentService txConsistentService(
         if (eventScannerEnabled) {
           new EventScanner(scheduler,
               eventRepository, commandRepository, timeoutRepository,
-              omegaCallback, eventPollingInterval).run();
+              omegaCallback, eventPollingInterval, nodeType).run();
```
[the remark this answers is not in the document]
Comment (coolbeevip) [reply alpha/alpha-server/src/main/java/org/apache/servicecomb/pack/alpha/server/AlphaConfig.java]: In the masterCheck method of the ClusterLockService class

**Pull request closed without merging**